### DISCUSSION TITLE
Confusion regarding the license – compared to Apache-2.0

### DISCUSSION THREAD

**Discussion by silverhook** [ImageMagick/ImageMagick#2758] [General] [answered]
Hi, I know ImageMagick does not hide the fact that its main license is based on Apache-2.0, but after I ran a diff and read through it, I would really welcome having some light shed on this.

Namely, I found only two (arguably three) changes to the license, of which I would say only one matters much:

- ImageMagick introduces a 69-line preamble
- ImageMagick, omits “(except as stated in this section)” from the _§3. Grant of Patent License_
- (and the name of the license is different)

Preamble (and license name) do not really influence the license itself, so the only actual difference between the two is an ambiguity in the patent license grant clause.

To better show the difference, here is §3 in Apache-2.0 (emphasis mine to show the difference between the two):

> **3. Grant of Patent License.** Subject to the terms and conditions of this License, each Contributor hereby grants to You a perpetual, worldwide, non-exclusive, no-charge, royalty-free, irrevocable _(except as stated in this section)_ patent license to make, have made, use, offer to sell, sell, import, and otherwise transfer the Work, where such license applies only to those patent claims licensable by such Contributor that are necessarily infringed by their Contribution(s) alone or by combination of their Contribution(s) with the Work to which such Contribution(s) was submitted. If You institute patent litigation against any entity (including a cross-claim or counterclaim in a lawsuit) alleging that the Work or a Contribution incorporated within the Work constitutes direct or contributory patent infringement, then any patent licenses granted to You under this License for that Work shall terminate as of the date such litigation is filed. 
 
…and same §3 in ImageMagic:

> **3. Grant of Patent License.** Subject to the terms and conditions of this License, each Contributor hereby grants to You a perpetual, worldwide, non-exclusive, no-charge, royalty-free, irrevocable patent license to make, have made, use, offer to sell, sell, import, and otherwise transfer the Work, where such license applies only to those patent claims licensable by such Contributor that are necessarily infringed by their Contribution(s) alone or by combination of their Contribution(s) with the Work to which such Contribution(s) was submitted. If You institute patent litigation against any entity (including a cross-claim or counterclaim in a lawsuit) alleging that the Work or a Contribution incorporated within the Work constitutes direct or contributory patent infringement, then any patent licenses granted to You under this License for that Work shall terminate as of the date such litigation is filed. 

What baffles me is that I do not see any reason why that part should be omitted. At best this change has no effect, at worst it means that ImageMagic license tries to get rid of the protection that the patent license in Apache-2.0 gives, while confusingly still leaving the actual text of the exception in.

I see no benefit in this change for either ImageMagick Studio LLC or the wider ImageMagick community, so any explanation why this is not simply an Apache-2.0 license would be greatly appreciated.

Looking forward to the discussion :)

**Comment by urban-warrior** [comment, marked as answer]
The ImageMagick license was updated to match the Apache 2.0 license excepting ownership to ImageMagick Studio LLC rather than the Apache organization.   If you spot additional differences now, please let us know.

**Reply by silverhook** [reply]
I see that this was fixed in https://github.com/ImageMagick/ImageMagick/commit/45e5d2493c08e7cb49f7268c01d847e88f78fd6c . Thank you.

**Comment by silverhook** [comment]
BTW, as an aside, I see you are bumping the year in the copyright notice. That is neither required by law, nor does it offer much in practical terms.

I wrote a [blog post about why bumping the year is not needed](http://matija.suklje.name/drafts/how-and-why-to-properly-write-copyright-statements-in-your-code#why-not-bump-the-year-on-change) and where this false idea (most likely) originated, as well as some more copyright-notice-related info.

**Comment by urban-warrior** [comment]
We only bump the copyright year to satisfy our OCD, not so much for legal reasons.
